src/gateway/GatewayV1.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.13;
+
+import "./Gateway.sol";
+
+interface IBalancerFlashLoan {
+    function flashLoan(
+        address recipient,
+        IERC20[] memory tokens,
+        uint256[] memory amounts,
+        bytes memory userData
+    ) external;
+}
+
+/// @title ECG Gateway V1
+/// @notice Gateway to interract via multicall with the ECG
+/// Owner can select which calls are allowed
+/// @custom:feature flashloan from balancer vault
+contract GatewayV1 is Gateway {
+    /// @notice Address of the Balancer Vault, used for initiating flash loans.
+    address public immutable balancerVault =
+        0xBA12222222228d8Ba445958a75a0704d566BF2C8;
+
+    function multicallWithBalancerFlashLoan(
+        IERC20[] calldata tokens,
```

Comment:
try to use native types (`address` here) for all public-facing storage variables, function args, and return values, it makes interfacing with other files / projects easier.